extensions/libraries/redcore/translation/helper.php

```diff
@@ -789,4 +789,142 @@ public static  function setLanguage($language, $loadLanguage = true)
+							}							
+						}
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Checks if a translation element for redITEM should be translatable on the current page.
+	 * It's necessary because of the way redITEM's custom fields are set up.
+	 * If the database structure of the redITEM component changes, the code will have to be changed too.
+	 *
+	 * @param   string  $table       The database table of the translation element
+	 * @param   array   $primarykey  The name of the columns of the translation element's reference IDs
+	 * @param   int     $itemid      The id of the current item being shown
+	 *                             
+	 * @return  void 
+	 */
+	public static function isReditemElementTranslatable($table, $primaryKeys, $itemid, $elementName)
+	{
+		//Check whether there's a relation between the current item and the translation element
+		$db = JFactory::getDbo();
+		$query = $db->getQuery(true)
+		->select($db->qn($primaryKeys[0]))
+		->from($db->qn($table))
+		->where($db->qn($primaryKeys[0]) . '=' . $db->q($itemid));
+
+		$db->setQuery($query);
+		$results = $db->loadObjectList();
+ 
+ 		//If there is, render a modal button & window in the toolbar
+		if (!empty($results))
+		{
+			$linkname = JText::_('LIB_REDCORE_TRANSLATION_NAME_BUTTON') . ' ' . $elementName;
+			$contentelement = str_replace('#__', '', $table);
+
+			self::renderTranslationModal($itemid, $linkname, $contentelement);
+		}		
+	}
+
+	/**
+	 * Renders a modal button & window for a translation element 
+	 *
+	 * @param   string  $itemid             The id of the current item being shown
+	 * @param   array   $linkname           The text to be shown on the modal button
+	 * @param   int     $contentelement     The current translation element
+	 * @param   string  $containingelement  The containing element that the modal button should be appended to
+	 *                             
+	 * @return  void
+	 */
+	public static function renderTranslationModal($itemid, $linkname, $contentelement)
+	{
+		RHelperAsset::load('full-page-modal.min.css', 'redcore');
+
+		if (!empty($itemid))
+		{
+			$modal = RLayoutHelper::render(
+				'modal.iframe-full-page',
+				array(
+					'options' => array(
+						'id' => $itemid,
+						'header' => '',
+						'linkName' => $linkname,
+						'link' => JRoute::_('index.php?option=com_redcore&view=translation&task=translation.edit&layout=modal-edit&contentelement=' . $contentelement . '&id=' . $itemid . '&tmpl=component'),
+						'linkClass' => 'btn btn-primary',
+					)
+				)
+			);
+		}
+		else 
+		{
+			$modal = '<button type="button" class="btn btn-primary disabled">' . JText::_('LIB_REDCORE_TRANSLATION_SAVE_ITEM_TO_TRANSLATE') . '</button>';
+		}
+		
+
+		//Render modal button & window, and move it to the end of the toolbar
+		echo '<div class="modalContainer" style="display:none;">' . $modal . '</div>';
+		$content = "jQuery( document ).ready(function() {
+						jQuery('.btn-toolbar').first().append(jQuery('.modalContainer'));
```

Comment:
this should not be hardcoded. This option belongs to the content element XML definition. By default it can be '.btn-toolbar' but you should not rely on that.